cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,342 @@
+	return fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", clusterName)
+}
+
+func generateDockerConfigJsonSecretConfigTo(name string, namespace string, clusterName string) secretbootstrap.SecretContext {
+	return secretbootstrap.SecretContext{
+		Cluster:   clusterName,
+		Name:      name,
+		Namespace: namespace,
+		Type:      "kubernetes.io/dockerconfigjson",
+	}
+}
+
+func updatePodScalerSecret(c *secretbootstrap.Config, o options) error {
+	key := fmt.Sprintf("%s.%s", o.clusterName, config)
+	return appendSecretItemContext(c, podScaler, string(api.ClusterAPPCI), key, secretbootstrap.ItemContext{
+		Field: serviceAccountKubeconfigPath(podScaler, o.clusterName),
+		Item:  podScaler,
+	})
+}
+
+func updateDPTPControllerManagerSecret(c *secretbootstrap.Config, o options) error {
+	const DPTPControllerManager = "dptp-controller-manager"
+	keyAndField := serviceAccountKubeconfigPath(DPTPControllerManager, o.clusterName)
+	return appendSecretItemContext(c, DPTPControllerManager, string(api.ClusterAPPCI), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  buildUFarm,
+	})
+}
+
+func updateRehearseSecret(c *secretbootstrap.Config, o options) error {
+	keyAndField := serviceAccountKubeconfigPath(ciOperator, o.clusterName)
+	return appendSecretItemContext(c, "pj-rehearse", string(api.ClusterBuild01), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  buildUFarm,
+	})
+}
+
+func updateChatBotSecret(c *secretbootstrap.Config, o options) error {
+	const chatBot = "ci-chat-bot"
+	name := chatBot + "-kubeconfigs"
+	keyAndField := serviceAccountKubeconfigPath(chatBot, o.clusterName)
+	return appendSecretItemContext(c, name, string(api.ClusterAPPCI), keyAndField, secretbootstrap.ItemContext{
+		Field: keyAndField,
+		Item:  chatBot,
+	})
+}
+
+func appendSecretItemContext(c *secretbootstrap.Config, name string, cluster string, key string, value secretbootstrap.ItemContext) error {
+	logrus.Infof("Appending secret item to: {name: %s, cluster: %s}", name, cluster)
+	sc, err := findSecretConfig(name, cluster, c.Secrets)
+	if err != nil {
+		return err
+	}
+	sc.From[key] = value
+	return nil
+}
+
+func updateExistingRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) error {
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) && cluster != string(api.ClusterARM01) && cluster != o.clusterName {
+			return appendRegistrySecretItemContext(c, regPullCredsAll, cluster, secretbootstrap.DockerConfigJSONData{
+				AuthField:   fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", o.clusterName),
+				Item:        buildUFarm,
+				RegistryURL: registryUrlFor(o.clusterName),
+			})
+		}
+	}
+	return nil
+}
+
+func generateRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) {
+	items := []secretbootstrap.DockerConfigJSONData{
+		{
+			AuthField:   "token_image-puller_ci_reg_auth_value.txt",
```

Comment:
`registryPullTokenField`

Reply:
Its a bit of an odd usage as I have to pass the clusterName as `ci`, but I think its fine to use that here.